tests/integration/cocoon/test_seed_sample_data.py

```diff
@@ -0,0 +1,102 @@
+import os
+import datetime
+import pytz
+import random
+
+logger = logging.getLogger()
+
+
+class CocoonTestPortfolioGroup(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        cls.scope = create_scope_id()
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+        cls.sample_data = pd.read_csv(
+            os.path.join("data", "seed_sample_data", "sample_data.csv")
+        )
+
+        seed_sample_data(
+            cls.api_factory,
+            ["portfolios", "instruments", "transactions"],
+            cls.scope,
+            os.path.join("data", "seed_sample_data", "sample_data.csv"),
+        )
+
+    def test_transactions_from_response(self):
+
+        transactions_from_response = self.api_factory.build(
+            lusid.api.TransactionPortfoliosApi
+        ).get_transactions(
+            scope=self.scope,
+            code=self.sample_data["fund_code"].to_list()[0],
+            from_transaction_date=datetime.datetime.now(tz=pytz.UTC)
+            - datetime.timedelta(weeks=52),
+        )
+
+        self.assertEqual(
+            set([txn.transaction_id for txn in transactions_from_response.values]),
+            set(self.sample_data["txn_id"].to_list()),
+        )
+
+    def test_portfolio_from_response(self):
+
+        portfolio_from_response = self.api_factory.build(
+            lusid.api.PortfoliosApi
+        ).get_portfolio(
+            scope=self.scope, code=self.sample_data["fund_code"].to_list()[0]
+        )
+
+        self.assertEqual(
+            portfolio_from_response.id.code, self.sample_data["fund_code"].to_list()[0]
+        )
+
+    def test_bad_file_type(self):
+
+        with self.assertRaises(ValueError) as error:
+
+            seed_sample_data(
+                self.api_factory,
+                ["bad_file_type"],
+                self.scope,
+                os.path.join("data", "seed_sample_data", "sample_data.csv"),
+            )
+
+        self.assertEqual(
+            error.exception.args[0],
+            "The provided file_type of bad_file_type has no associated mapping",
+        )
+
+    def test_instruments_from_response(self):
+
+        # In this test we take a random sample from the dataframe.
+        # We need to make a seperate get_instrument request for each item in the sample, so the test might become
+        # inefficient if we take the entire dataframe.
+        # A sample of approximately 10 should prove the function works as expected.
+
+        random_10_ids = set(
+            random.choices(self.sample_data["txn_instrument_id"].to_list(), k=10)
+        )
+        response_from_random_10 = set(
+            [
+                response.identifiers["ClientInternal"]
+                for response in [
+                    self.api_factory.build(lusid.api.InstrumentsApi).get_instrument(
```

Comment:
might be better to take the building of the api class outside the loop

Reply:
Good idea. Done.